app/mailers/user_mailer.rb

```diff
@@ -8,68 +8,35 @@ class UserMailer < ActionMailer::Base
     default from: AppConfig.first.admin_email, cc: AppConfig.first.admin_email
   end
 
-  def checkin_receipt(reservation)
+  # checks the status of the current reservation and sends the appropriate email
+  def reservation_status_update(reservation, receipt = false) # rubocop:disable all
     set_app_config
-    @reservation = reservation
-    mail(to: reservation.reserver.email,
-         subject: '[Reservations] Your equipment return receipt')
-  end
-
-  def checkout_receipt(reservation)
-    set_app_config
-    @reservation = reservation
-    mail(to: reservation.reserver.email,
-         subject: '[Reservations] Your equipment checkout receipt')
-  end
-
-  def missed_reservation_notification(reservation)
-    set_app_config
-    @reservation = reservation
-    mail(to: reservation.reserver.email,
-         subject: '[Reservations] Reservation Missed')
-  end
-
-  def overdue_checkin_notification(reservation)
-    set_app_config
-    @reservation = reservation
-    mail(to: reservation.reserver.email,
-         subject: '[Reservations] OVERDUE: equipment checkin')
-  end
 
-  def overdue_checked_in_fine(overdue_checked_in)
-    return if overdue_checked_in.equipment_model.late_fee == 0
-    set_app_config
-    @overdue_checked_in = overdue_checked_in
-    mail(to: overdue_checked_in.reserver.email,
-         subject: '[Reservations] Overdue equipment fine')
-  end
-
-  def reservation_confirmation(complete_reservation)
-    set_app_config
-    @complete_reservation = complete_reservation
-    mail(to: complete_reservation.first.reserver.email,
-         subject: '[Reservations] Confirmation of your reservation')
-  end
-
-  def upcoming_checkin_notification(reservation)
-    set_app_config
-    @reservation = reservation
-    mail(to: reservation.reserver.email,
-         subject: '[Reservations] Reminder: equipment check in')
-  end
+    return if reservation.status == 'returned overdue' &&
+              reservation.equipment_model.late_fee == 0
 
-  def request_approved_notification(reservation)
-    set_app_config
-    @reservation = reservation
-    mail(to: reservation.reserver.email,
-         subject: '[Reservations] Reservation request approved!')
-  end
-
-  def request_denied_notification(reservation)
-    set_app_config
     @reservation = reservation
-    mail(to: reservation.reserver.email,
-         subject: '[Reservations] Reservation request denied')
+    @status = @reservation.status
+
+    if @reservation.start_date == Time.zone.today &&
+       @status == 'reserved'
+      @status = 'starts today'
+    elsif @reservation.due_date == Time.zone.today &&
+          @status == 'checked out'
+      @status = 'due today'
+    end
+
+    if receipt && (@status == 'due today' || @status == 'overdue')
```

Comment:
No I understand that, what I'm saying is that someone could end up issuing a request to the `get_receipt` route, passing a reservation that isn't yet checked out and therefore it should fail. Unless I missed it, we're not currently checking for that anywhere (so if you end up calling `reservation_status_update(res, true)` and `res.checked_out == nil` the e-mail view will probably fail since it will try to call `strftime` on `nil`). Does that make more sense?